enterprise-suite/templates/es-console.yaml

```diff
@@ -72,22 +79,100 @@ data:
     server {
       listen 8080;
       server_name localhost;
+      set $base /usr/share/nginx/html;
 
-      location / {
-        root  /usr/share/nginx/html;
-        index index.html index.htm;
-        try_files $uri$args $uri$args/ /index.html;
+      # use external resolver to lookup backends, cache for 30 seconds
+
+      resolver 127.0.0.1:5353 ipv6=off valid=30s;
+
+      set $prometheus "prometheus-server.{{.Release.Namespace}}.svc.cluster.local.";
+      set $monitorapi "es-monitor-api.{{.Release.Namespace}}.svc.cluster.local.";
+      set $grafana "grafana-server.{{.Release.Namespace}}.svc.cluster.local.";
+
+      # proxy_pass backends with dynamic base url
+      # (redirects to ' $1/' with a space prefix tells nginx to not add server and port)
+
+      sub_filter_once off;
+
+      location ~ (.*/service/prometheus)(/.*) {
+        proxy_pass http://$prometheus$2$is_args$args;
+        sub_filter '="/' '="$1/';
+        sub_filter 'PATH_PREFIX = ""' 'PATH_PREFIX = "$1"';
+        proxy_redirect '/' ' $1/';
+      }
+
+      location ~ (.*/service/es-monitor-api)(/.*) {
+        proxy_pass http://$monitorapi$2$is_args$args;
+        proxy_redirect '/' ' $1/';
+      }
+
+      location ~ (.*/service/grafana)(/.*) {
+        proxy_pass http://$grafana:3000$2$is_args$args;
```

Comment:
It'd be nicer if we could use service discovery for the port, rather than hardcoding it. Not a super big deal though. It's available in the dns srv record and env vars, but I'm guessing nginx can't easily use either.

Reply:
I have no idea why we defined a service on a funky port instead of port 80. It might be a Grafana-ism, which wants to write fully qualified things in places which might include the port.

Or it might be someone copy and pasted random things from the internet.... :-|

Reply:
@katsutoxin @srikanthops do either of you know why we picked port 3000 for Grafana, instead of port 80 like everything else?

Reply:
@dbrinegar  port 3000 is the default grafana port.  Please refer http://docs.grafana.org/installation/configuration/#http-port